src/worker.js

```diff
@@ -23,7 +23,6 @@ var buffer; // All pthreads share the same Emscripten HEAP as SharedArrayBuffer
 var DYNAMICTOP_PTR = 0;
 var DYNAMIC_BASE = 0;
 
-var ENVIRONMENT_IS_PTHREAD = true;
```

Comment:
is this not needed? it's outside the MODULARIZE scope for that mode, which maybe does need this?

Reply:
Instead of just having it as a var, I thought the best way to handle this was to set `Module['ENVIRONMENT_IS_PTHREAD']` in the thread, and in `shell_pthreads.js` set `var ENVIRONMENT_IS_PTHREAD` based off `Module.ENVIRONMENT_IS_PTHREAD`. That way you don't get weird scoping issues, and `ENVIRONMENT_IS_PTHREAD` is explicitly set.
I think it makes sense to do this for all pthreads code, so it's not a module specific change.

Reply:
Oh, so something similar to the src/preamble.js change in https://github.com/emscripten-core/emscripten/pull/8365/files ? Yeah, I think that direction makes sense.

Reply:
Yeah, but this case it's only passing a bool around, and only on initialisation, so I think overhead would be negligible.